lib/msal-angular/src/msal.interceptor.config.ts

```diff
@@ -11,6 +11,11 @@ export declare type MsalInterceptorAuthRequest = Omit<PopupRequest, "scopes"> |
 
 export type MsalInterceptorConfiguration = {
     interactionType: InteractionType.Popup | InteractionType.Redirect;
-    protectedResourceMap: Map<string, Array<string> | null>;
+    protectedResourceMap: Map<string, Array<string|ProtectedResourceScopes> | null>;
     authRequest?: MsalInterceptorAuthRequest | ((msalService: MsalService, req: HttpRequest<unknown>, originalAuthRequest: MsalInterceptorAuthRequest) => MsalInterceptorAuthRequest);
 };
+
+export type ProtectedResourceScopes = {
```

Comment:
Open to suggestions of a different shape for this. Another possibility could be `type ProtectedResourceScopes = Record<string, string[]>`, which would support:
```javascript
["http://applicationD.com", [
  "all.scope",
  {"GET": ["read.scope"]}
]]
```

Reply:
I'll think about this overnight, but I think what you have here is good, as it allows us to easily add more things, and the typings will be more helpful.

Reply:
Curious to hear other's thoughts, but I'm cool with the object shape defined here.

Reply:
I'm a fan of this approach